src/main/java/com/conveyal/r5/streets/StreetLayer.java

```diff
@@ -1170,16 +1170,16 @@ public int getOrCreateVertexNear(double lat, double lon, StreetMode streetMode)
         // The split is somewhere along a street away from an existing intersection vertex. Make a new splitter vertex.
         int newVertexIndex = vertexStore.addVertexFixed((int) split.fixedLat, (int) split.fixedLon);
         int oldToVertex = edge.getToVertex(); // Hold a copy of the to vertex index, because it may be modified below.
+        ArrayList<int[]> geoms = new ArrayList<>();
         if (edge.isMutable()) {
             // The edge we are going to split is mutable.
             // We're either building a baseline graph, or modifying an edge created within the same scenario.
             // Modify the existing bidirectional edge pair to serve as the first segment leading up to the split point.
             // Its spatial index entry is still valid, since the edge's envelope will only shrink.
             edge.setLengthMm(split.distance0_mm);
             edge.setToVertex(newVertexIndex);
-            // Turn the edge into a straight line.
-            // FIXME split edges and new edges should have geometries!
-            edge.setGeometry(Collections.EMPTY_LIST);
+            geoms = edge.splitGeometryAfter(split.seg);
```

Comment:
It looks like we split the geometry and assign the split geometry to the new edges only when the original edge is mutable. The else clause below would result in straight-line geometries for splits in scenarios. I think once we've reached this section of code (`// The split is somewhere along a street away from an existing intersection vertex`) we always need to split any existing intermediate geometry, so rather than making a `new ArrayList<>()` above, we can just always perform the split with `ArrayList<int[]> geoms = edge.splitGeometryAfter(split.seg)`, then use the returned split geometries differently in these two different branches of the conditional.